src/edu/upenn/cit594/datamanagement/JSONFileReader.java

```diff
@@ -13,9 +13,10 @@ public JSONFileReader(String fileName) {
         super(fileName);
     }
 
-    public T read() {
+    public DataStore<T> read() {
         Object obj = null;
-
+        if(dataStore != null) return dataStore;
```

Comment:
This check becomes redundant with our `getReaderData` or `getExistingData` method in the `Processor`. I like we are putting in protection mechanisms, but we can choose either putting this protection in the `Reader` or having the special get method in the `Processor`. We should have this protection mechanism written only once. What do you think?

Reply:
I would advocate removing this and keeping the special get method in the `Processor`. Makes it more apparent to TAs what we are trying to do vs. this one line that might get lost

Reply:
Makes sense to me to only have it in one place. Which do you think I should take out?

I think having the memoization in Processor has a clearer intent. They might not notice or understand it in the Reader. Plus the Reader is for reading, so makes less sense for it to have this optimization.

Reply:
I agree!